bs_detector/bias.py

```diff
@@ -475,27 +483,59 @@ def print_raw_data_for_features(list_of_sentences):
             print(feat.keys())
             KEYS_DONE = True
         print(feat.values())
-#print_raw_data_for_features(get_list_from_file('input_text_original'))
+
+
+# print_raw_data_for_features(get_list_from_file('input_text_original'))
 
 def compute_bias(sentence_text):
-    features = extract_bias_features(unicode(sentence_text, errors='ignore'))
+    if isinstance(sentence_text, unicode):
+        features = extract_bias_features(sentence_text)
+    else:
+        features = extract_bias_features(unicode(sentence_text, errors='ignore'))
     BS_SCORE = (-0.5581467 +
-          0.3477007 * features['vader_sentiment'] +
-          -2.0461103 * features['opinion_rto'] +
-          0.5164345 * features['modality'] +
-          8.3551389 * features['liwc_3pp_rto'] +
-          4.5965115 * features['liwc_tent_rto'] +
-          5.737545 * features['liwc_achiev_rto'] +
-          5.6573254 * features['liwc_discr_rto'] +
-          -0.953181 * features['bias_rto'] +
-          9.811681 * features['liwc_work_rto'] +
-          -16.6359498 * features['factive_rto'] +
-          3.059548 * features['hedge_rto'] +
-          -3.5770891 * features['assertive_rto'] +
-          5.0959142 * features['subj_strong_rto'] +
-          4.872367 * features['subj_weak_rto'])
+                0.3477007 * features['vader_sentiment'] +
+                -2.0461103 * features['opinion_rto'] +
+                0.5164345 * features['modality'] +
+                8.3551389 * features['liwc_3pp_rto'] +
+                4.5965115 * features['liwc_tent_rto'] +
+                5.737545 * features['liwc_achiev_rto'] +
+                5.6573254 * features['liwc_discr_rto'] +
+                -0.953181 * features['bias_rto'] +
+                9.811681 * features['liwc_work_rto'] +
+                -16.6359498 * features['factive_rto'] +
+                3.059548 * features['hedge_rto'] +
+                -3.5770891 * features['assertive_rto'] +
+                5.0959142 * features['subj_strong_rto'] +
+                4.872367 * features['subj_weak_rto'])
     return BS_SCORE
 
+
+@contextmanager
+def poolcontext(*args, **kwargs):
+    pool = multiprocessing.Pool(*args, **kwargs)
+    yield pool
+    pool.terminate()
+
+
+def compute_statement_bias(statement_text, n_jobs=1):
+    sentences = nltk.sent_tokenize(statement_text.decode("ascii", "ignore"))
+    max_len = max(map(len, sentences))
+
+    if len(sentences) == 0:
+        return 0
+
+    avg_bias = 0
+
+    with poolcontext(processes=n_jobs) as pool:
+        bs_scores = pool.map(compute_bias, sentences)
+        avg_bias = sum(bs_scores)
+
+    if len(sentences) > 0:
+        avg_bias = round(float(avg_bias) / float(len(sentences)), 4)
```

Comment:
Why are we rounding to 4 digits? Is that a significant figures thing? Do we not trust the 1e-5 component to be accurate?

Reply:
This is keeping in line with @cjhutto's earlier code where round is also used. How do you feel about those lines too? I'm not tied to doing it, just wanting to keep in line with earlier convention.

Reply:
Yeah it is good to stick with convention @cjhutto why where you rounding? Just for making pretty tables?

Reply:
The input for bias ratings are integers -- four decimal places is waaaay more than enough precision.
Oh, and yes...Pretty Tables :)